### DISCUSSION TITLE
Farewell to the .NET Foundation

### DISCUSSION THREAD

**Discussion by MikeCodesDotNET** [AvaloniaUI/Avalonia#14666] [General]
In April 2020, Avalonia proudly joined the .NET Foundation, marking a significant milestone in our journey. Our joining the Foundation was a moment of great optimism, as we hoped to contribute to and benefit from a larger community of like-minded developers and projects. Over the years, Avalonia has flourished, becoming one of the most popular community projects within the Foundation, driven by the dedication and skill of our team and vibrant community.

As Avalonia and the .NET Foundation have evolved, it's become essential to evaluate the nature of this partnership and how it aligns with our goals. While we appreciate the .NET Foundation's mission to support open-source .NET projects, we've found our interactions with the Foundation to be extremely limited and haven't seen any significant benefits of being a member project. 

With this in mind, we have decided to leave the .NET Foundation. By returning ownership of Avalonia to its core team, we reaffirm our long-term dedication to its success. We have submitted an "Intent to Leave" with the .NET Foundation, and the process to leave has begun, with the expectation that we will have concluded our time as a member project by early April. 

### What does this mean for our community? 

While significant in principle, our departure from the .NET Foundation will have minimal practical impact on our ongoing operations. The primary change will involve transitioning our Contributor License Agreement (CLA) bot and moving Azure resources to our subscriptions. These adjustments are minor and will not disrupt our work or the community's involvement.

Beyond these specific changes, it will be business as usual for Avalonia. Our development, community interaction, and project management will continue as before. The essence of Avalonia remains unchanged.

The critical difference post-departure is that Avalonia will revert to being wholly owned and governed by those who have consistently invested in its growth and success. This transition ensures that the project remains in the hands of genuinely passionate individuals with a vested interest in the project's future. 

### FAQ

**Why did Avalonia decide to leave the .NET Foundation?**
We decided to leave the .NET Foundation as we believe it's right that the project returns to the hands of those who are genuinely passionate about the project and have a vested interest in its future.

**Will this change affect Avalonia's open-source status?**
No. Avalonia will remain open-source. We are committed to keeping Avalonia freely available for all developers. 

**Are there any plans to change how Avalonia interacts with its contributors or community?** 
No. The change in relationship with the .NET Foundation is primarily a legal and structural change concerning the project's ownership. It does not entail any changes to interaction and engagement strategies with our community and contributors. We remain committed to maintaining open and collaborative communication with our community, ensuring our relationship continues as before.
   
**Are there any plans to join other foundations or organisations?**
No. We have no plans to join any other organisations. 

**How will Avalonia's financial investments change following the departure?**
The .NET Foundation does not provide any financial support to projects; thus, no changes will occur due to leaving the Foundation. 

AvaloniaUI OÜ, the company formed by the core team, will continue to invest in Avalonia's future. In 2023, AvaloniaUI OÜ invested more than half a million euros into the development of Avalonia. We expect this investment to increase year-on-year. 

**How was the decision to leave the .NET Foundation made?**
The decision to leave the .NET Foundation was reached following extensive internal discussions that spanned over a year. This was not a decision made in haste but rather a considered response to our ongoing experiences with the Foundation. Crucially, a consensus was reached unanimously by the core team, reflecting our collective vision and agreement on the best path forward for Avalonia. 

**What was Avalonia looking for in terms of support from the .NET Foundation?**
When we joined the .NET Foundation, our aspirations were to gain support in areas where Avalonia needed help. Understanding that financial aid wasn't part of the Foundation's offerings, we were optimistic about receiving guidance and assistance in documentation, marketing strategies, and general advice on managing a popular open-source project. Expertise in these areas is crucial for the growth and sustainability of any popular OSS project.

However, our experience was contrary to these expectations, as we encountered a notable lack of interaction and support from the Foundation. This absence of engagement led us to reassess the value that the Foundation was adding to our project. 

Signed by: @grokys
Signed by: @kekekeks 
Signed by: @danwalmsley

**Comment by thevortexcloud** [comment]
> However, our experience was contrary to these expectations, as we encountered a notable lack of interaction and support from the Foundation. This absence of engagement led us to reassess the value that the Foundation was adding to our project.

[This seems to be a recurring problem...](https://www.glennwatson.net/posts/dnf-problems-solutions)

**Reply by TeddyAlbina** [reply]
Yeah the foundation need to be more proactive

**Comment by JonDouglas** [comment]
Thank you for being so transparent about this decision, Mike!

**Comment by edandersen** [comment]
Probably worth doing before your repos are transferred out like what happened in 2021.

**Comment by TETYYS** [comment]
Makes perfect sense, why would Avalonia give ownership to the project to someone without anything in return?

**Comment by robloo** [comment]
Great writeup. Makes sense considering how things have evolved with the .NET foundation over the years. Everyone was more optimistic there would be more value back when the foundation was first getting started. Now there seems to be some clear risks, yet no value other than some compute resources. I think everyone can understand this decision.

**Comment by Hooterr** [comment]
Microsoft is very keen on taking credit for other people's work...

**Reply by bawkee** [reply]
Unfortunately, Microsoft rarely even contributes to projects that Windows 11 uses all over the place (WinUI3) so this is not surprising. Avalonia is currently the only .NET UI framework that, at least in my experience as a fulltime desktop dev, has a very bright future.

**Reply by dongle-the-gadget** [reply]
WinUI 3 is maintained by Microsoft themselves.

**Reply by SuperJMN** [reply]
> WinUI 3 is maintained by Microsoft themselves.

Only thing they maintain in that project is the steering wheel right to the cliff.

**Comment by kuiperzone** [comment]
Am I correct in understanding that the .NET Foundation owns Avalonia. I see from your MIT license that the copyright is theirs. How are you going to "return ownership"?

I guess you could simply license your own work under MIT and own further changes.

Over the years, I've become careful about ownership myself, esp in regard to open source. I know you guys have done a great job and really wish you the best.

**Reply by thevortexcloud** [reply]
The only way to relicense all the existing code would be to get the existing rights holders to agree to it. On some projects that can mean contacting literally every single person who has contributed to the project and hoping they agree (there are some fairly large projects that have successfully done this, but it takes several years in some cases). But in this case they just have to convince the dotnet foundation I guess? In which case I am curious who would gain ownership once it has been "returned".  Does it revert to all the contributors or is going to Avalonia OU?

**Reply by kuiperzone** [reply]
I never mentioned "re-licensing" it. I was simply suggesting that MIT is extremely permissive, to the point of near public domain. Even if .NET Foundation has any claim on ownership, then there is nothing stopping Avalonia using and developing its own work under the terms of MIT itself, as anyone can do anyway irrespective of "ownership". As I see things, "ownership" (as in control) of an MIT licensed project rests largely on recognition and the ability to sustain that by keeping the project active. This is something large corporations can do, but hard for individuals and small teams who have other things in life (such as earning a living).

In any case, the decision is up to the Avalonia team and its for them to decide if, indeed, these comments are even relevant. If I may contribute a suggestion, it would be to consider "ownership" (in terms of who controls the project), and consider a copy left element to future source, while keeping it relatively unrestricted in terms of general use.

Like I say, I'm really supportive of Avalonia and want to see it succeed and be sustainable.

**Reply by kuiperzone** [reply]
And to add something on a different point here, while I have not contributed to the source itself, I have contributed to Avalonia open source eco-system. But more than, I am in the process of founding a business built around Avalonia and my partner and I have discussed making donations to Avalonia in the future. But who would we be funding in terms of who reaps the rewards? I would be happy to support a plucky and creative underdog such as Avalonia, but would have been upset to have learned that I have essentially given money for the ultimate benefit of MS.

**Comment by jimfoye** [comment]
`let surpriseFactor = 0 // not mutable`

**Comment by workgroupengineering** [comment]
someone from Avalonia could apply https://twitter.com/1kevgriff/status/1780613769611071867?t=Wla8lvBqk9gYPkK2KiN3hA&s=19

**Reply by MikeCodesDotNET** [reply]
No one on our team has the time or energy available to become a board member of the DNF, but perhaps a prominent community member (like yourself) should apply.

**Reply by workgroupengineering** [reply]
If I had the capability, I would propose to support Avalonia.

**Reply by 1kevgriff** [reply]
I fully support your candidacy @workgroupengineering

**Reply by workgroupengineering** [reply]
Thanks, but there are many more competent people than me, like @robloo

**Comment by sgf** [comment]
The .net Foundation does not appear to be able to play any significant role, and it might be wise to withdraw from it.

An important factor for open source projects is financial support. As a foundation, if it cannot provide appropriate help, then what is the point of the foundation's existence? So the existence of the .net Foundation is a joke.
